staging/src/k8s.io/apimachinery/pkg/util/wait/wait.go

```diff
@@ -36,6 +37,15 @@ var ForeverTestTimeout = time.Second * 30
 // NeverStop may be passed to Until to make it never stop.
 var NeverStop <-chan struct{} = make(chan struct{})
 
+// StartUntil starts f in a new goroutine and calls done once f has finished.
+func StartUntil(stopCh <-chan struct{}, wg *sync.WaitGroup, f func(stopCh <-chan struct{})) {
```

Comment:
Yes, the name is odd. Open to suggestions :)
`f` has this specific signature because it is the signature that is used most of the time. I want the refactored code that uses this helper function to look cleanly and that's why the signature is like this - to avoid anonymous functions which clutter the code unnecessarily (in this case IMO). Consider how it will look with anonymous closures - not as nice, right?
If we go for an anonymous function we might as well just remove this helper function and write the whole snippet every time because there is not much value in it then. I tried to extract the common bit :)
I'll improve the comment once/if there is an agreement this is the way forward.